### ISSUE TITLE
[Bug]: Optional dep handling not consistent with pnpm

### ISSUE THREAD

**Issue by matthewjh** [aspect-build/rules_js#1199]
### What happened?

In pnpm, if an `optional` dep's lifecycle hook fails, the failure is swallowed and the hook marked as "skipped". The package source is still linked into node_modules, and the overall installation is allowed to proceeed:
https://github.com/pnpm/pnpm/blob/main/exec/build-modules/src/index.ts#L151C28-L151C28

With rules_js, a lifecycle hook error bubbles up and fails the build, even for optional dependencies. There is a `no_optional` attr on `npm_translate_lock`, but this skips every aspect of optional dependencies installation (i.e. does not create the links and bazel targets), rather than merely allowing the hooks to fail.

This is important for us, and I imagine others, as some packages have install hooks that install or build native binaries that are not supported on all platforms (e.g. mac arm). On these platforms we want to allow the hook to fail, and the install or build to proceed, on the understanding certain aspects of that package will not operate at runtime on those platforms.

### Version

Development (host) and target OS/architectures:

Output of `bazel --version`:

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:

Language(s) and/or frameworks involved:

**Comment by alexeagle** [comment]
Yeah that sounds like a legit bug to me, we should be fully compatible with pnpm behavior.

**Comment by jbedard** [comment]
@matthewjh are you still having this issue? Are you working around it somehow? Or maybe it is not as big of a deal with modern pnpm?
